### PR TITLE
perf(artifacts): Set a limit for number of history keys to query for artifact run history 

### PR DESCRIPTION
**Pull request by ibindlish**
[PR here](https://github.com/wandb/core/pull/17101) adds a limit parameter to the run history endpoint. 

This PR updates the `artifactVersionRunHistoryRow` op to use that parameter to limit the number of keys returned while fetching the history for the artifact pages (model registry version table, artifact version table and artifact metadata tab)

follow up app PR: https://github.com/wandb/core/pull/17123

### CODE REVIEW

File: weave-js/src/core/ops/domain/artifactVersion.ts [hunk 1]
```diff
@@ -638,6 +638,7 @@ export const opArtifactVersionRunHistoryRow = makeBasicOp({
   name: 'artifactVersion-historyMetrics',
   argTypes: {
     artifactVersion: TypeHelpers.nullableOneOrMany('artifactVersion' as const),
+    maxKeyLimit: 'number' as const,
```
Comment (ibindlish) [inline weave-js/src/core/ops/domain/artifactVersion.ts:641]: How do I make these params optional?

Comment (estellazx) [reply weave-js/src/core/ops/domain/artifactVersion.ts:641]: does `maxKeyLimit: TypeHelpers.maybe('number' as const)` work?

Comment (ibindlish) [reply weave-js/src/core/ops/domain/artifactVersion.ts:641]: hmm, just tried this - doesn't seem to work. I can't use the `opArtifactVersionRunHistoryRow` op without specifying a number for the maxKeyLimit argument

Comment (ibindlish) [reply weave-js/src/core/ops/domain/artifactVersion.ts]: nevermind, it worked!

**Comment by ibindlish** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ibindlish** [comment]
> **CLA Assistant Lite bot:** Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/wandb/weave/blob/master/cla/cla.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.
> 
> I have read the CLA Document and I hereby sign the CLA
> 
> You can retrigger this bot by commenting **recheck** in this Pull Request

recheck

**Review by tssweeney — APPROVED** [review]

**Pull request merged**